source/lib/ze_lib.cpp

```diff
@@ -46,19 +53,48 @@ namespace ze_lib
         std::string loaderFullLibraryPath = create_library_path(MAKE_LIBRARY_NAME( "ze_loader", L0_LOADER_VERSION), loaderLibraryPath.c_str());
         loader = LOAD_DRIVER_LIBRARY(loaderFullLibraryPath.c_str());
 
-        if( NULL == loader )
+        if( NULL == loader ) {
+            printf("loader load failed\n");
             return ZE_RESULT_ERROR_UNINITIALIZED;
+        }
 
         typedef ze_result_t (ZE_APICALL *loaderInit_t)();
         auto loaderInit = reinterpret_cast<loaderInit_t>(
                 GET_FUNCTION_PTR(loader, "zeLoaderInit") );
+        printf("calling loader init static\n");
         result = loaderInit();
-        if( ZE_RESULT_SUCCESS == result ) {
-            typedef HMODULE (ZE_APICALL *getTracing_t)();
-            auto getTracing = reinterpret_cast<getTracing_t>(
-                GET_FUNCTION_PTR(loader, "zeLoaderGetTracingHandle") );
-            tracing_lib = getTracing();
+        if( ZE_RESULT_SUCCESS != result ) {
+            return result;
+        }
+        typedef HMODULE (ZE_APICALL *getTracing_t)();
+        auto getTracing = reinterpret_cast<getTracing_t>(
+            GET_FUNCTION_PTR(loader, "zeLoaderGetTracingHandle") );
+        if (getTracing == nullptr) {
+            printf("missing getTracing\n");
+            return ZE_RESULT_ERROR_UNINITIALIZED;
+        }
+        tracing_lib = getTracing();
+        typedef ze_result_t (ZE_APICALL *zelLoaderDriverCheck_t)(ze_init_flags_t flags, ze_init_driver_type_desc_t* desc, ze_global_dditable_t *globalInitStored, zes_global_dditable_t *sysmanGlobalInitStored, bool *requireDdiReinit, bool sysmanOnly);
+        auto loaderDriverCheck = reinterpret_cast<zelLoaderDriverCheck_t>(
+                GET_FUNCTION_PTR(loader, "zelLoaderDriverCheck") );
+        if (loaderDriverCheck == nullptr) {
+            printf("missing loaderDriverCheck\n");
+            return ZE_RESULT_ERROR_UNINITIALIZED;
         }
+        typedef ze_result_t (ZE_APICALL *zelLoaderTracingLayerInit_t)(std::atomic<ze_dditable_t *> &zeDdiTable);
+        auto loaderTracingLayerInit = reinterpret_cast<zelLoaderTracingLayerInit_t>(
+                GET_FUNCTION_PTR(loader, "zelLoaderTracingLayerInit") );
+        if (loaderTracingLayerInit == nullptr) {
+            printf("missing loaderTracingLayerInit\n");
```

Comment:
I think these should be going to stderr

Reply:
agreed, I was going to convert the printfs to debug trace logs for the ones we determine useful.

Reply:
Fixed in the latest patches.